### DISCUSSION TITLE
How to set periodic BC?

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#23261] [Q&A General] [answered]
Hi everyone,

Do you know how I can set the upper layer and bottom layer's boundary condition as periodic BC?

Thank you,
Best

**Comment by abc-hy** [comment]
I mean, the upper boundary and bottom boundary

**Reply by GiudGiud** [reply]
There s a periodic BC object you can try.

as long as the surfaces are well aligned it will work

**Reply by abc-hy** [reply]
![image](https://user-images.githubusercontent.com/63679626/214957625-81b79074-e161-4b50-938a-87cd1536dd82.png)
I tried, there is no PeriodicBC object...Which object do you refer to?

**Reply by GiudGiud** [reply]
You ll have to find the exact name in the source index

**Comment by ngrilli** [comment, marked as answer]
@abc-hy  I normally use the auto_direction option to set periodic boundary conditions
between the top and bottom boundaries, see an example here:

https://github.com/ngrilli/c_pfor_am/blob/main/test/tests/SandfeldZaiser2015/SandfeldLoop.i

**Reply by abc-hy** [reply]
Thank you for your reply.
I want to set the BC of top and bottom boundary as periodic and remain the left and right as zero flux boundary condition.\
I wrote a command as below:
![image](https://user-images.githubusercontent.com/63679626/214980746-0de06540-deb0-46b2-8d5f-5d5d19791946.png)
Is this correct?

**Reply by GiudGiud** [reply]
This should work as long as top & bottom match rigorously
